### DISCUSSION TITLE
Should tea consider nested environments?

### DISCUSSION THREAD

**Discussion by jakobvase** [pkgxdev/pkgx#723] [Ideas]
In #681, I described how I moved from a root folder in a project with one environment requirement (node 14) to a subfolder with another environment requirement (node 16). It seems pretty clear to me that tea should pick up this new environment on a `cd subfolder/`.

However, there are also dependencies that are shared between these two environments. The root environment also has an `export AWS_PROFILE=my-profile` and a dependency on yarn 1.22. Should these also be used for the inner environment?

`nvm` doesn't care about the root environment in this case, but it also doesn't have to. It only cares about node, so it only needs to check for versions of that. They don't have this problem. I don't know much about other environment managers (rbenv etc.).

The options are:

1. Don't consider nested environments. Only go up the tree to the first place where environments are specified (eg. a tea.yaml or package.json exists), activate packages defined there and stop.

2. Consider nested environments. Go up the tree to the git root and activate the packages specified along the way, that are not already specified.

**Comment by jakobvase** [comment]
Personally, I prefer the first option. I like that it's very explicit, even though we as developers will have to copy paste things more. It gives us more control. The second option feels a bit more like css, and css can be a mess to navigate in (why is this style applied here?). But if done just right, the second option is more powerful. There just sort of has to be a way to un-depend on parent packages then, right?

**Comment by rawkode** [comment]
I'd really love to see option two supported, this would be incredibly useful for monorepositories as we don't need to duplicate our dependencies for each project within the repository.

The best approach is to enable this behaviour by default, finding all pkgx.yaml files up until `.git`

If we're running dev outside of a Git repository, don't allow the behaviour?

@mxcl I see you closed the issue, so I'm commenting here. Do let me know if there's a discussion about this that I've not found.
